Spatial_Engine/Query/IsSelfIntersecting.cs

```diff
@@ -69,6 +80,20 @@ public static bool IsSelfIntersecting(this IElement2D element2D, double toleranc
             return false;
         }
 
+
+        /******************************************/
+        /****   Public Methods - Interfaces    ****/
+        /******************************************/
+
+        [Description("Returns if the one dimensional representation of the IElement is closer to itself than the tolerance at any two points.")]
+        [Input("element", "The IElement to evaluate self intersections from.")]
+        [Input("tolerance", "Minimum distance to be considered intersecting.", typeof(Length))]
+        [Output("A boolean which is true if the IElement is self intersecting.")]
```

Comment:
```suggestion
        [Description("Returns if the one dimensional representation of an IElement is closer to itself than the tolerance at any two points.")]
        [Input("element", "The IElement to evaluate self intersections from.")]
        [Input("tolerance", "Minimum distance to be considered intersecting.", typeof(Length))]
        [Output("A boolean which is true if the IElement is self intersecting.")]
```
I'm not sure about that one but is "one dimensional" suitable for `IElement2D`s? Probably yes, if we consider outlines as `IElement1D`s.
Also I think if we go the through interface we might lose an important information from `IElement2D` description ("Does not check for intersections between external and internal curves, or between different internal curves").

Reply:
Good point, I will add an extra definition explaining how does the check work for the IElement2D. One-dimensional representation makes similar sense to me as to you - I see it as _wireframe_ of the multi-dimensional object.